lib/common-types.ts

```diff
@@ -221,6 +221,19 @@ export interface MetadataProps {
   tags: Link<'Tag'>[]
 }
 
+export type MetadataAggeratedRootProps = {
+  metadata: {
+    withReferences: [
+      {
+        entity: { sys: { type: 'Link'; linkType: 'Entry'; id: 'my-page-1' } }
+        filter: {
+          'sys.contentType.metadata.annotations.ContentType[nin]': ['Contentful:AggregateRoot']
+        }
+      }
+    ]
+  }
+}
```

Comment:
This still includes hardcoded values from the example payload. We need to map this into types, something like:
```
export type MetadataAggeratedRootProps = {
  metadata: {
    withReferences: {
       entity: Link<Entry>,   // explanation: whatever is passed as entity matches the Link type for Entry
       filter: Record<ScheduledActionReferenceFilters, string[]>   // explanation: whatever is passed in the filter is an object with keys that are included in the `ScheduledActionReferenceFilters` enum and values an array of strings
    }[]   // the [] here means that withReferences is an Array
}
```

Also `ScheduledActionsReferenceFilters` doesn't have a very flexible name and we cannot change it now, maybe you can create an alias and use that one here instead, like this:
`export type ReleasesReferenceFilters = ScheduledActionsReferenceFilters`

Reply:
Oh okay thanks! I'll update it too